django-cloudlaunch/cloudlaunch/backend_plugins/base_vm_app.py

```diff
@@ -265,27 +323,35 @@ def launch_app(self, provider, task, name, cloud_config,
         vm_type = cloudlaunch_config.get(
             'vmType', cloud_config.get('default_instance_type'))
 
-        log.debug("Launching with subnet %s and VM firewalls %s" %
-                  (subnet, vmfl))
-        log.info("Launching base_vm of type %s with UD:\n%s" % (vm_type,
-                                                                user_data))
-        task.update_state(state='PROGRESSING',
-                          meta={'action': "Launching an instance of type %s "
-                                "with keypair %s in zone %s" %
-                                (vm_type, kp.name, placement_zone)})
+        log.debug("Launching with subnet %s and VM firewalls %s", subnet, vmfl)
+
+        if provider_config.get('ssh_public_key'):
+            # cloud-init config to allow login w/ the config ssh key
+            # http://cloudinit.readthedocs.io/en/latest/topics/examples.html
+            log.info("Adding a cloud-init config public ssh key to user data")
+            user_data += """
+#cloud-config
+ssh_authorized_keys:
```

Comment:
Is it allowed to have multiple ssh_authorized_keys sections in user_data? Just wondering in case user_data already contains user supplied keys, but probably not something we need to necessarily address right now.

Reply:
Agree. Given we've never used this in the past let's just see how things go.